File: ocaml/raindrops/raindrops.ml
open Base

let raindrop n = 
  match n % 3, n % 5, n % 7 with
  | (0, 0, 0) -> "PlingPlangPlong"
  | (0, 0, _) -> "PlingPlang"
  | (0, _, 0) -> "PlingPlong"
  | (_, 0, 0) -> "PlangPlong"
  | (0, _, _) -> "Pling"
  | (_, 0, _) -> "Plang"
  | (_, _, 0) -> "Plong"
  | _ -> Int.to_string n

File: lua/raindrops/raindrops.lua
---@param n number
---@return string
return function(n)
    if n % 3 == 0 and n % 5 == 0 and n % 7 == 0 then
        return 'PlingPlangPlong'
    elseif n % 3 == 0 and n % 5 == 0 then
        return 'PlingPlang'
    elseif n % 5 == 0 and n % 7 == 0 then
        return 'PlangPlong'
    elseif n % 3 == 0 and n % 7 == 0 then
        return 'PlingPlong'
    elseif n % 3 == 0 then
        return 'Pling'
    elseif n % 5 == 0 then
        return 'Plang'
    elseif n % 7 == 0 then
        return 'Plong'
    end

    return tostring(n)
end
